### DISCUSSION TITLE
Predefined compile-option for Sonoff Zigbee-Bridge (75)

### DISCUSSION THREAD

**Discussion by thomas-lentz** [arendst/Tasmota#21338] [Support request] [answered]
Hi,
there are some predefined compile-options for sensors, displays, ...
Is there also one for the ESP8266-based Sonoff Zigbee-Bridge? Some Background: I'm using this device outside, near the seaside. I wanted to be sure, that the case I'm using is really waterproof and so I integrated a SHT31 device connected to the I2C of the Sonoff Zigbee-Bridge.
Unfortunately the SHT31 is recognized (i2cdevice found at 0x44), but there is no supporting i2cdriver compiled with the standard [Zigbee-Bridge Tasmota firmware](http://ota.tasmota.com/tasmota/release/tasmota-zbbridge.bin).
Is it enough to enable **CUSTOM_CONFIG_ZIGBEE** ([see Q&A #21060](https://github.com/arendst/Tasmota/discussions/21060)) in the **user_config_override.h** (and the necessary i2cdriver for the SHT31) to get an Ziebee-Bridge image just like **tasmota-zbbridge**.bin?
I appreciate your help
 Tom

**Comment by HofeBY** [comment]
What is displayed with your current Image by command "I2Cdriver" ?

**Reply by thomas-lentz** [reply]
07:11:10.399 MQT: stat/xx/zigbee/RESULT = {"I2CDriver":""}
No driver ist installed

**Comment by sfromis** [comment]
To build for ESP8266-based Sonoff Zigbee Bridge, see build environment `tasmota-zbbridge`, which is also selecting the board definition to match the hardware with 2 MB of flash.

Beware that you should **not** select `FIRMWARE_ZBBRIDGE` unless you need to build that exact setup, without any amendments, meaning no sensor drivers added. Here's the resulting standard set of `#define`, and especially `#undef` which makes that an unworkable plan:
https://github.com/arendst/Tasmota/blob/development/tasmota/include/tasmota_configurations.h#L521-L683
At the end of that block, you can see the `#define` done for including that type of Zigbee support.

That `CUSTOM_CONFIG_ZIGBEE` is not in the Tasmota source code, just an example environment from that thread, meant for copying.

**Reply by thomas-lentz** [reply, marked as answer]
Hi sfromis,
thanks for your reply. This is exactly what I was searching for. I understod everything you said, except:

> especially #undef which makes that an unworkable plan:

I added the following line to the _my_user_config.h_
`//#define FIRMWARE_ZBBRIDGE                        // Create tasmota-zbbridge for sonoff ZB-Bridge (ESP8266)`


then I added to the _tasmota_configurations.h_ the line
`#define USE_SHT3X                              // [I2cDriver15] Enable SHT3x (I2C address 0x44 or 0x45) or SHTC3 (I2C address 0x70) sensor (+0k7 code)`

after the

`//#undef USE_I2C                                   // Enable I2C, zbbridge uses i2c EEprom`

line in the:
```
/*********************************************************************************************\
 * [tasmota-zbbridge.bin]
 * Provide an image for the Sonoff Zigbee Bridge
\*********************************************************************************************/
```
section.

After that I compiled with _platformio run -e tasmota-zbbridge_ and did an firmware upload. 
The nice thing is, that all Zigbee sensor configuration and also the general Tasmota configuration were still "alive" (I upgraded from 13.4.0 (stock firmware) to 13.4.1.2.

Here is what it looks like:
![image](https://github.com/arendst/Tasmota/assets/53970628/fa533174-8d1b-4b5a-89ab-9f601de21674)

Kind regards
 Tom

**Reply by sfromis** [reply]
You are not "supposed" to modify `tasmota_configurations.h` at all, but keep modifications in one of the `override` files. This strategy works better when updating source, so that you do not have to repeat the same modifications.

Hence the recommendation is that you create your own build environment, based on `tasmota32`, which means that you should not use `FIRMWARE_ZBBRIDGE` (a fixed environment), and what I said about `#undef` is to highlight the fact that the standard structure does not allow you to add features to, or subtract features from those *fixed* environments. Following the intended strategy does not include this.
https://tasmota.github.io/docs/Compile-your-build/#customize-your-build

Of course, you can do whatever you want, and which works for you. I just wanted to point out how customization is meant to work.

**Comment by jonny190** [comment]
Has anyone got any newer firmware then 6.7.8.0 build 373 or know how to build newer? as the lateset zigbee2mqtt is no longer supported

**Reply by sfromis** [reply]
I'm having 6.7.9 installed, as supplied by the Tasmota project:
https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp
Not at all "new", but a higher number than 6.7.8.

In practice, my Zigbee setup is primarily using a bridge with CC2652 instead.
